src/buffer.rs

```diff
@@ -30,23 +30,34 @@ pub struct Buffer {
 }
 
 // TODO: need to actually make a Theme dict lol
+/// Styles a string accordingly to the string and returns a Span
+///
+/// * text - The text that will be styled into a [`Span`]
+/// * token - The token name, expected to be returned from [`Node`].kind()
 fn write_token<'a, T>(text: T, token: &'static str) -> Span<'a>
 where
 	T: Into<Cow<'a, str>>,
 {
 	Span::styled(
 		text,
-		Style::default().fg(match token {
-			"function" => Color::Rgb(246, 199, 255),
-			"identifier" => Color::Cyan,
-			"string" => Color::Yellow,
-			"comment" => Color::Green,
-			_ => Color::White,
-		}),
+		Style::default().fg(
+			// TODO: we really need to make the theme into a dict of some sort
+			match token {
+				"function" => Color::Rgb(246, 199, 255),
+				"identifier" => Color::Cyan,
+				"string" => Color::Yellow,
+				"comment" => Color::Green,
+				_ => Color::White,
+			},
+		),
 	)
 }
 
-fn clamp(v: usize, x: usize, y: usize) -> usize {
+/// Clamps a value between two other values
+fn clamp<T>(v: T, x: T, y: T) -> T
```

Comment:
worth noting that there's an `Ord::clamp` in the stdlib that lets you write `v.clamp(x, y)`